PackageExplorer/PackageViewer.xaml.cs

```diff
@@ -489,13 +470,109 @@ private void PackageFolderContextMenu_Opened(object sender, RoutedEventArgs e)
             menu.Opened -= PackageFolderContextMenu_Opened;
         }
 
+        private void OnTreeViewItemCopy(object sender, ExecutedRoutedEventArgs e)
+        {
+            if (sender is TreeView treeView && treeView.SelectedItem is PackagePart packagePart)
+            {
+                var data = CreateDataObject(packagePart);
+                Clipboard.SetDataObject(data);
+            }
+        }
+
+        private void OnTreeViewItemPaste(object sender, ExecutedRoutedEventArgs e)
+        {
+            PackageFolder folder = null;
+
+            if (sender is TreeView treeView)
+            {
+                folder = treeView.SelectedItem as PackageFolder;
+            }
+
+            if (HandleDataObject(folder, Clipboard.GetDataObject(), true))
+            {
+                e.Handled = true;
+            }
+        }
+
+        private void OnDataContextChanged(object sender, DependencyPropertyChangedEventArgs e)
+        {
+            if (Clipboard.ContainsData(PackageFileDataFormat))
+            {
+                Clipboard.Clear();
+            }
+        }
+
+        private bool HandleDataObject(PackageFolder folder, IDataObject data, bool copy)
+        {
+            if (DataContext is PackageViewModel model)
+            {
+                if (model.IsSigned || model.IsInEditFileMode || model.IsInEditMetadataMode)
+                {
+                    return false;
+                }
+            }
+
+            if (data.GetDataPresent(PackageFileDataFormat))
+            {
+                if (data.GetData(PackageFileDataFormat) is string packagePartPath)
+                {
+                    folder = folder ?? RootFolder;
+
+                    var packagePart = RootFolder.GetPackageParts().FirstOrDefault(part => part.Path == packagePartPath);
+
+                    if (packagePart != null)
```

Comment:
This and other things currently prevent drag & drop / copy & past from one NPE instance into another one.